controller/src/main/java/io/pravega/controller/store/stream/ZKStreamMetadataStore.java

```diff
@@ -202,6 +203,12 @@ ZKScope newScope(final String scopeName) {
         return storeHelper.checkExists(stream.getStreamPath());
     }
 
+    @Override
+    public CompletableFuture<Integer> getSafeStartingSegmentNumberFor(final String scopeName, final String streamName) {
+        return storeHelper.getData(String.format(ZKStoreHelper.DELETED_STREAMS_PATH, getScopedStreamName(scopeName, streamName)))
+                          .handleAsync((data, ex) -> (ex == null) ? BitConverter.readInt(data.getData(), 0) + 1 : 0);
```

Comment:
only for DataNotFoundException should we ignore and default to 0. For everything else, we should log and throw the exception.

Reply:
Thanks, now the method logs and throws `CompletionException(ex)` in case of an exception different from `StoreException.DataNotFoundException`.